### PR TITLE
Use project_id in delete projectjoin mutation

### PR DESCRIPTION
**Pull request by k9845**
Addresses 
- https://github.com/the-deep/server/issues/949
- https://github.com/the-deep/server/issues/948

## Changes
- Use project_id instead of project_join_request_id in deleteprojectjoin mutation

Mention related users here if any.

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] n+1 queries
- [x] flake8 issues
- [x] `print`
- [x] typos
- [x] unwanted comments

## This PR contains valid:

- [x] tests
- [ ] permission checks (tests here too)
- [ ] translations

### CODE REVIEW

File: apps/project/mutation.py [hunk 1]
```diff
@@ -47,21 +47,24 @@ class Arguments:
...
     result = graphene.Field(ProjectJoinRequestType)
 
     @staticmethod
-    def mutate(root, info, id):
+    def mutate(root, info, project_id):
         try:
             instance = ProjectJoinRequest.objects.get(requested_by=info.context.user,
                                                       status=ProjectJoinRequest.Status.PENDING,
-                                                      id=id)
+                                                      project=project_id)
```
Comment (thenav56) [inline apps/project/mutation.py:61]: Let's make sure users can't create more than one ProjectJoinRequest.
We can make this validation on a database level to using unique_together (project_id, user_id)

**Review by thenav56 — CHANGES_REQUESTED** [review]

File: apps/analysis_framework/schema.py [hunk 2]
```diff
@@ -97,6 +98,7 @@ class AnalysisFrameworkDetailType(AnalysisFrameworkType):
     secondary_tagging = DjangoListField(WidgetType)  # Without section
     members = DjangoListField(AnalysisFrameworkMembershipType)
     preview_image = graphene.Field(FileField)
+    visible_projects = DjangoListField(ProjectType)
```
Comment (thenav56) [inline apps/analysis_framework/schema.py]: We may need to create a strict project type for this.
cc: @AdityaKhatri

File: apps/analysis_framework/dataloaders.py [hunk 3]
```diff
@@ -60,6 +62,17 @@ def batch_load_fn(self, keys):
         return Promise.resolve([_map[key] for key in keys])
 
 
+class VisibleProjects(DataLoaderWithContext):
+    def batch_load_fn(self, keys):
+        project_qs = Project.objects\
+            .filter(analysis_framework__in=keys)\
+            .exclude(models.Q(is_private=True) & ~models.Q(members=self.context.request.user))
```
Comment (thenav56) [inline apps/analysis_framework/dataloaders.py]: Let's use `get_for_gq` to avoid repeating common logic.

**Review by thenav56 — CHANGES_REQUESTED** [review]

File: apps/analysis_framework/schema.py (hunk 2 above)
Comment (AdityaKhatri) [reply apps/analysis_framework/schema.py]: Yeah, let's do it.
Lets only expose `title`, `id` & `isPrivate`.

File: apps/analysis_framework/dataloaders.py [hunk 4]
```diff
@@ -60,6 +61,15 @@ def batch_load_fn(self, keys):
         return Promise.resolve([_map[key] for key in keys])
 
 
+class VisibleProjects(DataLoaderWithContext):
+    def batch_load_fn(self, keys):
+        project_qs = Project.get_for_gq(self.context.request.user)
```
Comment (thenav56) [inline apps/analysis_framework/dataloaders.py]: We don't need keys?

File: apps/project/schema.py [hunk 5]
```diff
@@ -121,6 +120,16 @@ def resolve_membership_pending(root, info):
         return info.context.dl.project.join_status.load(root.pk)
 
 
+class AnalysisFrameworkProjectType(DjangoObjectType):
```
Comment (thenav56) [inline apps/project/schema.py]:
```suggestion
class AnalysisFrameworkVisibleProjectType(DjangoObjectType):
```

File: apps/project/schema.py [hunk 6]
```diff
@@ -121,6 +120,16 @@ def resolve_membership_pending(root, info):
...
+    class Meta:
+        model = Project
+        fields = (
```
Comment (thenav56) [inline apps/project/schema.py:127]: skip_registry

**Review by thenav56 — CHANGES_REQUESTED** [review]

**Review by thenav56 — APPROVED** [review]

**Pull request merged**